### DISCUSSION TITLE
Scene Control Buttons Question

### DISCUSSION THREAD

**Discussion by RunnerMojo** [RavenSystem/esp-homekit-devices#1699] [Q&A]
I am implementing a Wemos D1 HAA device for scene control.  The idea is to use the Wemos with four GPIO buttons to implement a wall switch scene controller, but there are no relays or outputs in this device, it's only used to control other HomeKit devices.  I have tested and know I can do this with HomeKit automations by creating a Switch in the Wemos then using that to trigger an automation to control the various other HomeKit devices. 

My question is: How can I toggle the state of the target HomeKit device with each subsequent press of the Wemos Button?  So in that way, each of the four buttons can control one device with the simplest user interface for the less technically inclined.  Push-On, Push-Off, etc.  Apple doesn't seem to have a toggle function for other non HAA devices?
Thanks.

**Comment by wahidulislamriyad** [comment]
You can do this even without any HomePod.
Create your scene controller device as an accessory and control target devices by HomeKit API.
Example json to do this when you have enabled API:
    "a": [{
            "t": 2,
            "s": 5,
            "0": {
                "r": [{
                    "g": 3,
                    "v": 0
                }],
                "h": [{
                    "h": "homebridge.lan",
                    "p": 5556,
                    "m": 1,
                    "u": "characteristic",
                    "c": "{\"characteristics\":[{\"aid\":4,\"iid\":9,\"value\":false}]}"
                }]
            },
            "1": {
                "r": [{
                    "g": 3,
                    "v": 1
                }],
                "h": [{
                    "h": "homebridge.lan",
                    "p": 5556,
                    "m": 1,
                    "u": "characteristic",
                    "c": "{\"characteristics\":[{\"aid\":4,\"iid\":9,\"value\":true}]}"
                }]
            },
            "f0": [{
                "g": 1,
                "t": 1
            }],
            "f1": [{
                "g": 1,
                "t": 0
            }]
        }

**Reply by RunnerMojo** [reply]
I am looking to control accessories that I added to my network before HomeBridge or HAA.  Just commercial HomeKit devices such as a Vocolinc outlet.  When I use curl to dump the list of accessories and characteristics, this only shows me items I have connected to my HomeBridge.  But I don't see devices added directly to my LAN using HAA or standard commercial devices.  So if I was to implement the solution above, I would need to now the aid and iid of other devices.   Not sure how to get those.  
Thanks.
